storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -117,5 +199,103 @@ func (s *peersSource) Read(
 
 	wg.Wait()
 
+	close(incrementalQueue)
+	incrementalWg.Wait()
+
+	if incremental {
+		// Now cache the incremental results
+		shards := make([]uint32, 0, len(shardsTimeRanges))
+		for shard := range shardsTimeRanges {
+			shards = append(shards, shard)
+		}
+
+		if err := blockRetriever.CacheShardIndices(shards); err != nil {
+			return nil, err
+		}
+	}
+
 	return result, nil
 }
+
+func (s *peersSource) incrementalFlush(
+	namespace ts.ID,
+	persistManager persist.Manager,
+	shard uint32,
+	shardRetrieverMgr block.DatabaseShardBlockRetrieverManager,
+	shardResult result.ShardResult,
+	tr xtime.Range,
+) error {
+	var (
+		numSeries      = len(shardResult.AllSeries())
+		flushedBlocks  = make([]incrementalFlushedBlock, 0, numSeries)
```

Comment:
This is a pretty big allocation per call, can we pass it in and reuse it across multiple calls of `incrementalFlush` instead to reduce memory allocation and GC?